### ISSUE TITLE
Can BART in fairseq do text infilling? (one <mask> token to multiple tokens)

### ISSUE THREAD

**Issue by yeounyi** [pytorch/fairseq#3434]
I heard that ```denoising.py``` in fairseq can support text infilling in BART. 

By text infilling I mean such task like this.
```  
"UN Chief Says There Is No <mask> in Syria"
'UN Chief Says There Is No Plan to Stop Chemical Weapons in Syria'
```
Can you give me a sample code to do this task?

**Comment by varinf** [comment]
Yes it can. I would start with fine-tuning BART on some data where I replace a span with a single MASK token. In fact this is one of the experiment we did in ["Data Augmentation using Pre-trained Transformer Models"](https://github.com/varinf/TransformersDataAugmentation) Refer to the code [here](https://github.com/varinf/TransformersDataAugmentation/blob/main/src/bart_aug/denoised_dataset.py#L288 )
